### PR TITLE
chore: Commit message linting in the website repo

### PR DESCRIPTION
**Pull request by sobi3ch**

### CODE REVIEW

File: commitlint.config.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+        'style',
+
+        // test: add missing tests or correct existing tests
+        'test'
+      ]
+    ],
+
+    // scope is optional, when used it must be in lower case and have one of the below values.
+    'scope-case': [2, 'always', 'lower-case'],
+    'scope-enum': [2, 'always', ['c', 'elixir', 'go', 'javascript', 'rust', 'swift']],
```
Comment (mrinalwadhwa) [inline commitlint.config.js]: We can remove this role. These scopes don't apply to the website repo

File: commitlint.config.js [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        // feat(c): so we can later genrate implementation specific changelogs
+        //
+        // If a commit affects multiple implementations, please break it into two commits.
+        'feat',
+
+        // fix: a bug fix
+        //
+        // It is generally a good idea to add a implementation scope to a bug fix commit
+        // fix(c): so we can later genrate implementation specific changelogs
+        //
```
Comment (mrinalwadhwa) [inline commitlint.config.js]: we can remove these lines.

File: commitlint.config.js [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+        'ci',
+
+        // docs: a documentation only change
+        'docs',
+
+        // feat: a new feature
+        //
+        // It is generally a good idea to add a implementation scope to a feature commit
+        // feat(c): so we can later genrate implementation specific changelogs
+        //
```
Comment (mrinalwadhwa) [inline commitlint.config.js]: we can remove this lines

File: commitlint.config.js [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+        // refactor: code change that neither fixes a bug nor adds a feature
+        'refactor',
+
+        // style: changes that do not affect the meaning of the code (white-space, formatting etc.)
+        'style',
+
+        // test: add missing tests or correct existing tests
+        'test'
+      ]
```
Comment (mrinalwadhwa) [inline commitlint.config.js]: please add another type here call `content:`

**Comment by mrinalwadhwa** [comment]
@sobi3ch thanks for starting this, added some comments.

Adding @miedziak @mattgreg as reviewers to the rules as they contribute to this repo frequently. Please review the rules.

We can customize the type and scope enums best fit the needs of this repo. I think a separation between code and content of the website would be helpful, hence the suggestion for a `content:` type.

What do you all think?

File: commitlint.config.js (hunk 1 above)
Comment (sobi3ch) [reply commitlint.config.js]: done

File: commitlint.config.js (hunk 2 above)
Comment (sobi3ch) [reply commitlint.config.js]: done

File: commitlint.config.js (hunk 4 above)
Comment (sobi3ch) [reply commitlint.config.js]: done

File: commitlint.config.js (hunk 3 above)
Comment (sobi3ch) [reply commitlint.config.js]: done

**Comment by sobi3ch** [comment]
> @sobi3ch thanks for starting this, added some comments.
> 
> Adding @miedziak @mattgreg as reviewers to the rules as they contribute to this repo frequently. Please review the rules.
> 
> We can customize the type and scope enums best fit the needs of this repo. I think a separation between code and content of the website would be helpful, hence the suggestion for a `content:` type.
> 
> What do you all think?
`content` type sounds reasonable.

**Comment by mrinalwadhwa** [comment]
@miedziak could you merge this at the right time ... note that this linting forces all lower case commit message subject lines. So I suggest merging any other open PRs that use upper case first or this will cause their builds to fail.

**Comment by mrinalwadhwa** [comment]
just realized that the commit message of this commit itself wouldn't pass linting so that may need to be amended before a merge @miedziak @sobi3ch

**Comment by sobi3ch** [comment]
> just realized that the commit message of this commit itself wouldn't pass linting so that may need to be amended before a merge @miedziak @sobi3ch

corrected

**Pull request merged**